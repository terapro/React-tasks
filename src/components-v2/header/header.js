import React from 'react';
import {Logo} from 'src/components-v2/common/logo/logo';
import {Menu} from 'src/components-v2/menu/menu';

export const Header = () => (
  <div className="header">
    <Logo />
    <Menu />
  </div>
);
