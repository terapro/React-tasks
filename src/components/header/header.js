import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {openSearch} from "src/actions";
import {TopBar} from 'src/components/header/topbar/topbar.js';
import {Search} from 'src/components/search/search.js';
import {Film} from 'src/components/film/film.js'

const HeaderChild = ({filmMode, currentFilm, onOpenSearch, children}) => (
    <header className={'header'}>
      <div className={'header-container'}>

        <TopBar searchButtonClick={onOpenSearch}  filmMode={filmMode}/>
        {children}
          {/*<Search />
          <Film info={currentFilm} active ={filmMode} />*/}
      </div>
    </header>
);

HeaderChild.propTypes = {
    filmMode: PropTypes.bool,
    currentFilm: PropTypes.shape({
        id: PropTypes.number,
        title: PropTypes.string,
        tagline: PropTypes.string,
        vote_average: PropTypes.number,
        vote_count: PropTypes.number,
        release_date: PropTypes.string,
        poster_path: PropTypes.string,
        overview: PropTypes.string,
        budget: PropTypes.number,
        revenue: PropTypes.number,
        genres: PropTypes.arrayOf(PropTypes.string),
        runtime: PropTypes.number
    }),
    onOpenSearch: PropTypes.func.isRequired
};

HeaderChild.defaultProps = {
    filmMode: false,
    currentFilm: {}
};

export const Header = connect(
  store =>
    ({
      filmMode: store.mode.film,
      currentFilm: store.films.currentFilm
    }),
  dispatch =>
    ({
      onOpenSearch(){dispatch(
        openSearch()
      )}
    })
)(HeaderChild);
