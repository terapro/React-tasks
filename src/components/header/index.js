import React from 'react';
import PropTypes from 'prop-types';

import {TopBar} from 'src/components/header/topbar/topbar.js';
import {Search} from 'src/components/search/index.js';
import {Film} from 'src/components/film/film.js'


const Header = (props) => (
        <header className={'header'}>
          <div className={'header-container'}>
            <TopBar setSearchModeCallback = {props.setSearchModeCallback}  filmMode={props.filmMode}/>
            {props.searchMode? <Search startSearch = {props.startSearch} active={props.searchMode} /> : null }
            {props.filmMode? <Film info={props.filmInfo} active ={props.filmMode} /> : null}
          </div>
        </header>
);

Header.propTypes = {
  startSearch: PropTypes.func.isRequired,
  searchMode: PropTypes.bool,
  filmMode: PropTypes.bool,
  filmInfo: PropTypes.object,
  setSearchModeCallback: PropTypes.func
};
Header.defaultProps = {
  searchMode: true,
  filmMode: false,
  filmInfo: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
  }).isRequired,
  setSearchModeCallback: () => {}
};


export {Header};