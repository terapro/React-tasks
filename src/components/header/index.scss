.header {
  height: 250px;
  width: inherit;
  background: #000 url('img/netflix-bg.jpg')
}

@import 'container/container';
@import 'header-logo/header-logo';
@import 'header-search/header-search';
@import 'header-title/header-title';






