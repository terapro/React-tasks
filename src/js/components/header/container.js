import React from 'react';

import {HeaderLogo} from './header-logo.js';
import {HeaderTitle} from './header-title.js';
import {HeaderSearch} from './header-search.js';

const HeaderContainer = () => (
    <div className={'header-container'}>
        <HeaderLogo/>
        <HeaderTitle/>
        <HeaderSearch/>
    </div>

);

export {HeaderContainer};