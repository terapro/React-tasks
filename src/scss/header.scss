.header {
  height: 250px;
  width: inherit;
  background: #000 url('../img/netflix-bg.jpg')

}

.header-container {
  width: $headerContainerWidth;
  margin: 0 auto;
  padding: 20px 0;
}

.header-logo {

}

.header-title {
  margin-top: 20px;
}

.header-search {
  margin-top: 10px;
}

.search-components {
  display: flex;
  margin-top: 10px;
  justify-content: space-between;
  align-items: center;

}