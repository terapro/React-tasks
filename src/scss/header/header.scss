.header {
  background-color: red;
  height: 210px;
  width: inherit;
  background: #000 url('../img/netflix-bg.jpg')

}