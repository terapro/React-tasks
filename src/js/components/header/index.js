import React from 'react';

import {HeaderContainer} from './container.js';


const Header = () => (
        <header className={'header'}>
            <HeaderContainer/>
        </header>
);


export {Header};