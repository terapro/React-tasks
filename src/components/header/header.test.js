import React from 'react';
import {Header} from 'src/components/header/header.js';
import {shallow} from 'enzyme';

const mockFunc = jest.fn();

const mockFilm = {
  "id": 333339,
  "title": "Ready Player One",
  "tagline": "A better reality awaits.",
  "vote_average": 8.1,
  "vote_count": 617,
  "release_date": "2018-03-28",
  "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
  "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
  "budget": 175000000,
  "revenue": 0,
  "genres": [
    "Adventure",
    "Science Fiction",
    "Action"
  ],
  "runtime": 140
};

describe('<Header />', () => {
  describe('Rendering', () => {
    it('should render the component in search mode', () => {
      const wrapper = shallow(<Header startSearch={mockFunc} filmInfo={mockFilm} filmMode={false} searchMode />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component in film mode', () => {
      const wrapper = shallow(<Header startSearch={mockFunc} filmInfo={mockFilm} filmMode searchMode={false}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
