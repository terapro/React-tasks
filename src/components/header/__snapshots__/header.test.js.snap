// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Rendering should render the component in film mode 1`] = `
<header
  class="header"
>
  <div
    class="header-container"
  >
    <div
      class="topbar"
    >
      <img
        class="logo"
        src="img/logo-netflixroulette.png"
      />
    </div>
    <div
      class="header-search"
    >
      <h2
        class="ttl ttl-white ttl-UC"
      >
        Find your movie
      </h2>
      <input
        class="search-input"
        placeholder="Start typing..."
        type="text"
        value=""
      />
      <div
        class="search-components"
      >
        <div
          class="search-type-section"
        >
          <div
            class="lbl lbl-UC lbl-white"
          >
            search by
          </div>
          <div
            class="btn btn-red btn-UC"
            id=""
          >
            title
          </div>
          <div
            class="btn btn-grey btn-UC"
            id=""
          >
            genres
          </div>
        </div>
        <div>
          <div
            class="btn btn-big btn-red"
            id=""
          >
            Search
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`<Header /> Rendering should render the component in search mode 1`] = `
<header
  class="header"
>
  <div
    class="header-container"
  >
    <div
      class="topbar"
    >
      <img
        class="logo"
        src="img/logo-netflixroulette.png"
      />
    </div>
    <div
      class="header-search"
    >
      <h2
        class="ttl ttl-white ttl-UC"
      >
        Find your movie
      </h2>
      <input
        class="search-input"
        placeholder="Start typing..."
        type="text"
        value=""
      />
      <div
        class="search-components"
      >
        <div
          class="search-type-section"
        >
          <div
            class="lbl lbl-UC lbl-white"
          >
            search by
          </div>
          <div
            class="btn btn-red btn-UC"
            id=""
          >
            title
          </div>
          <div
            class="btn btn-grey btn-UC"
            id=""
          >
            genres
          </div>
        </div>
        <div>
          <div
            class="btn btn-big btn-red"
            id=""
          >
            Search
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`<Header /> Rendering should render the component with default props 1`] = `
<header
  class="header"
>
  <div
    class="header-container"
  >
    <div
      class="topbar"
    >
      <img
        class="logo"
        src="img/logo-netflixroulette.png"
      />
    </div>
    <div
      class="header-search"
    >
      <h2
        class="ttl ttl-white ttl-UC"
      >
        Find your movie
      </h2>
      <input
        class="search-input"
        placeholder="Start typing..."
        type="text"
        value=""
      />
      <div
        class="search-components"
      >
        <div
          class="search-type-section"
        >
          <div
            class="lbl lbl-UC lbl-white"
          >
            search by
          </div>
          <div
            class="btn btn-red btn-UC"
            id=""
          >
            title
          </div>
          <div
            class="btn btn-grey btn-UC"
            id=""
          >
            genres
          </div>
        </div>
        <div>
          <div
            class="btn btn-big btn-red"
            id=""
          >
            Search
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
